### PR TITLE
Clone aws images across regions

### PR DESCRIPTION
**Pull request by Gundersanne**
Adds 3 new endpoints. To clone images across regions:

1. post to /composes/{id}/clone
2. poll status at /clones/{id} until complete
3. list clones of a specific compose at /composes/{id}/clones

---

TODO 
- add  clone to aws integration test?
- what if it's a pure sharing thing (so without a new target region?); do we still treat it as any other clone and deduplicate the actual AMIs in the FE?

### CODE REVIEW

File: internal/db/db.go [hunk 1]
```diff
@@ -104,6 +109,29 @@ func (db *dB) GetCompose(jobId string, orgId string) (*ComposeEntry, error) {
...
+		return "", err
+	}
+	defer conn.Release()
+
+	result := conn.QueryRow(ctx, sqlGetComposeImageType, orgId, jobId)
+
+	var imageType string
+	err = result.Scan(&imageType)
+	if err != nil {
+		if errors.As(err, &pgx.ErrNoRows) {
```
Comment (kingsleyzissou) [inline internal/db/db.go:166]: Should this be `errors.Is` like it is here:
https://github.com/osbuild/osbuild-composer/commit/9def5455708f7e247fe4103d47e00a7beb52169b

File: internal/db/db.go [hunk 2]
```diff
@@ -193,3 +234,79 @@ func (db *dB) CountComposesSince(orgId string, duration time.Duration) (int, err
...
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Release()
+
+	var clone CloneEntry
+	err = conn.QueryRow(ctx, sqlGetClone, id, orgId).Scan(&clone.Id, &clone.Request, &clone.CreatedAt)
+	if err != nil {
+		if errors.As(err, &pgx.ErrNoRows) {
```
Comment (kingsleyzissou) [inline internal/db/db.go:304]: Same here

File: internal/v1/handler.go [hunk 3]
```diff
@@ -711,3 +711,195 @@ func buildCustomizations(cust *Customizations) *composer.Customizations {
...
+	if err != nil {
+		return err
+	}
+
+	idHeader, err := getIdentityHeader(ctx)
+	if err != nil {
+		return err
+	}
+
+	imageType, err := h.server.db.GetComposeImageType(composeId, idHeader.Identity.OrgID)
```
Comment (ondrejbudai) [inline internal/v1/handler.go:843]: I wonder if this check is even needed. Querying a compose other than an aws one does not harm. :shrug:

File: internal/v1/handler.go [hunk 4]
```diff
@@ -711,3 +711,195 @@ func buildCustomizations(cust *Customizations) *composer.Customizations {
...
+	}
+
+	offset := 0
+	if params.Offset != nil {
+		offset = *params.Offset
+	}
+
+	cloneEntries, count, err := h.server.db.GetClonesForCompose(composeId, idHeader.Identity.OrgID, limit, offset)
+	if err != nil {
+		return err
```
Comment (ondrejbudai) [inline internal/v1/handler.go:866]: I wonder whether it's valuable to translate this error. We do it for other DB queries but I'm not sure what's your rule of thumb.

Comment (Gundersanne) [reply internal/v1/handler.go:866]: ib is pretty bad at this yea, makes sense to always translate.

File: internal/v1/handler.go (hunk 3 above)
Comment (Gundersanne) [reply internal/v1/handler.go:843]: true

File: internal/db/migrations/007_alter_table_composes_alter_colum_request_json_to_jsonb.up.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+ALTER TABLE composes ALTER COLUMN request SET DATA TYPE jsonb;
```
[the remark this answers is not in the document]
Comment (Gundersanne) [reply internal/db/migrations/007_alter_table_composes_alter_colum_request_json_to_jsonb.up.sql:1]: I added it to a different commit instead. The one that actually this column as jsonb.

[the remark this answers is not in the document]
Comment (ondrejbudai) [reply internal/db/migrations/007_alter_table_composes_alter_colum_request_json_to_jsonb.up.sql:1]: Makes sense, thanks! :)

File: internal/db/db.go (hunk 2 above)
Comment (ondrejbudai) [reply internal/db/db.go:304]: I think this is still unresolved.

**Review by ondrejbudai — CHANGES_REQUESTED** [review]
One small mistake, otherwise, this is awesome! Thanks for fixing the commits and their messages. :)

File: internal/db/db.go (hunk 2 above)
Comment (kingsleyzissou) [reply internal/db/db.go:304]: Whoops I also just noticed that now too

Comment (Gundersanne) [reply internal/db/db.go:304]: Woops :facepalm:

**Review by kingsleyzissou — APPROVED** [review]
lgtm

**Review by ondrejbudai — APPROVED** [review]
:heart:

**Pull request merged**